src/net.cpp

```diff
@@ -2418,3 +2418,7 @@ void CNode::EndMessage() UNLOCK_FUNCTION(cs_vSend)
 
     return CSipHasher(k0, k1).Write(&vchNetGroup[0], vchNetGroup.size()).Finalize();
 }
+
+int64_t PoissonNextSend(int64_t nNow, int average_interval_seconds) {
+    return nNow + (int64_t)(log1p(GetRand(1ULL << 48) * -0.0000000000000035527136788 /* -1/2^48 */) * average_interval_seconds * -1000000.0 + 0.5);
```

Comment:
I am skeptical that this actually results in the given `average_interval_seconds`. I think based on [this](https://stephens999.github.io/fiveMinuteStats/inverse_transform_sampling.html#continuous_example:_exponential_distribution) that it produces an exponential distribution with mean twice what it should be.